src/psyclone/tests/psyir/transformations/intrinsics/minval2code_trans_test.py

```diff
@@ -0,0 +1,175 @@
+    array_ref = ArrayReference.create(
+        array_symbol, [Reference(i_iterator), Reference(j_iterator),
+                       Reference(k_iterator)])
+    result = trans._loop_body(True, array_iterators, var_symbol, array_ref)
+    assert result.debug_string() == (
+        "if (var(i,k) > array(i,j,k)) then\n"
+        "  var(i,k) = array(i,j,k)\n"
+        "end if\n")
+
+
+def test_init_var():
+    '''Test that the _init_var method works as expected.'''
+    trans = Minval2CodeTrans()
+    var_symbol = DataSymbol("var", REAL_TYPE)
+    result = trans._init_var(var_symbol)
+    # As 'huge' is not yet part of an expression, the 'debug_string()'
+    # method incorrectly assumes it is a call.
+    assert result.debug_string() == "call HUGE(var)\n"
+
+
+def test_str():
+    '''Test that the str method, implemented in the parent class, works
+    as expected.
+
+    '''
+    trans = Minval2CodeTrans()
+    assert str(trans) == ("Convert the PSyIR MINVAL intrinsic to equivalent "
+                          "PSyIR code.")
+
+
+def test_name():
+    '''Test that the name method, implemented in the parent class, works
+    as expected.
+
+    '''
+    trans = Minval2CodeTrans()
+    assert trans.name == "Minval2CodeTrans"
+
+
+def test_validate():
+    '''Test that the validate method, implemented in the parent class,
+    works as expected.
+
+    '''
+    trans = Minval2CodeTrans()
+    with pytest.raises(TransformationError) as info:
+        trans.validate(None)
+    assert ("Error in Minval2CodeTrans transformation. The supplied node "
+            "argument is not an intrinsic, found 'NoneType'."
+            in str(info.value))
+
+
+def test_apply(fortran_reader, fortran_writer):
+    '''Test that the apply method, implemented in the parent class, works
+    as expected.
+
+    '''
+    code = (
+        "subroutine minval_test(array,n,m)\n"
+        "  integer :: n, m\n"
+        "  real :: array(10,20)\n"
+        "  real :: result\n"
+        "  result = minval(array)\n"
+        "end subroutine\n")
+    expected = (
+        "subroutine minval_test(array, n, m)\n"
+        "  integer :: n\n  integer :: m\n"
+        "  real, dimension(10,20) :: array\n"
+        "  real :: result\n  real :: minval_var\n"
+        "  integer :: i_0\n  integer :: i_1\n\n"
+        "  minval_var = HUGE(minval_var)\n"
+        "  do i_1 = 1, 20, 1\n"
+        "    do i_0 = 1, 10, 1\n"
+        "      if (minval_var > array(i_0,i_1)) then\n"
+        "        minval_var = array(i_0,i_1)\n"
+        "      end if\n"
+        "    enddo\n"
+        "  enddo\n"
+        "  result = minval_var\n\n"
+        "end subroutine minval_test\n")
+    psyir = fortran_reader.psyir_from_source(code)
+    # FileContainer/Routine/Assignment/UnaryOperation
+    intrinsic_node = psyir.children[0].children[0].children[1]
+    trans = Minval2CodeTrans()
+    trans.apply(intrinsic_node)
+    result = fortran_writer(psyir)
+    assert result == expected
```

Comment:
Please add a compilation check.

Reply:
Added (and for the tests for maxval and sum).